helix-core/src/main/java/org/apache/helix/model/ExternalView.java

```diff
@@ -95,6 +105,32 @@ public String getResourceName() {
     return _record.getId();
   }
 
+  /**
+   * Get the resource group name
+   *
+   * @return the name of the resource group this resource belongs to.
+   */
+  public String getResourceGroupName() {
```

Comment:
RoutingTableProvider call this to determine whether it needs to cache all resource group partition/instance information.